packages/medusa/src/repositories/product.ts

```diff
@@ -96,6 +106,41 @@ export class ProductRepository extends Repository<Product> {
       )
     }
 
+    if (categories) {
```

Comment:
thought(non-blocking): As we add more resources related to products, the product repository will continue to grow in complexity and size. It would be great to investigate what it would take to extract some of this logic - potentially to the service layer - to separate the product domain from categories, discount conditions, etc. This will lead to additional queries to the DB because we will use services to retrieve each related resource, which will be a tradeoff.

Reply:
We could break this down into query objects, I like the idea. But this definitely needs to happen post typeorm merge. At this point, it will be a super mess 😄